File: src/main/java/eu/siacs/conversations/ui/activity/result/ScanQrCode.java
package eu.siacs.conversations.ui.activity.result;

import android.content.Context;
import android.content.Intent;
import androidx.activity.result.contract.ActivityResultContract;
import androidx.annotation.NonNull;
import com.google.common.base.Strings;
import eu.siacs.conversations.ui.ScanActivity;

public class ScanQrCode extends ActivityResultContract<Void, String> {
    @NonNull
    @Override
    public Intent createIntent(@NonNull final Context context, final Void unused) {
        final var intent = new Intent(context, ScanActivity.class);
        return intent;
    }

    @Override
    public String parseResult(final int resultCode, final Intent intent) {
        if (resultCode == ScanActivity.RESULT_OK && intent != null) {
            return Strings.nullToEmpty(intent.getStringExtra(ScanActivity.INTENT_EXTRA_RESULT));
        }
        return null;
    }
}

File: src/main/java/eu/siacs/conversations/ui/activity/result/ScanQrCode.kt
package eu.siacs.conversations.ui.activity.result

import android.content.Context
import android.content.Intent
import androidx.activity.result.contract.ActivityResultContract
import com.google.common.base.Strings
import eu.siacs.conversations.ui.ScanQrCodeActivity

class ScanQrCode : ActivityResultContract<Void?, String?>() {
    override fun createIntent(context: Context, input: Void?): Intent =
        Intent(context, ScanQrCodeActivity::class.java)

    override fun parseResult(resultCode: Int, intent: Intent?): String? {
        if (resultCode == ScanQrCodeActivity.RESULT_OK && intent != null) {
            return Strings.nullToEmpty(intent.getStringExtra(ScanQrCodeActivity.INTENT_EXTRA_RESULT))
        }
        return null
    }
}
